### PR TITLE
Example of using the Dart internationalization package in Flutter

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
This looks great.  @abarth and @Hixie should probably at least see this go by.

lgtm.

File: examples/stocks/lib/main.dart [hunk 1]
```diff
@@ -104,6 +107,9 @@ class StocksAppState extends State<StocksApp> {
   }
 }
 
-void main() {
-  runApp(new StocksApp());
+void main()
+{
```
Comment (abarth) [inline examples/stocks/lib/main.dart]: We don't usually have a line break here.

**Comment by sethladd** [comment]
Super cool. Thanks!!

**Comment by sethladd** [comment]
cc @alan-knight

**Comment by alan-knight** [comment]
Cool. I note that you can omit the "args" argument when it's empty. And the import path is wrong :-)

File: examples/stocks/lib/main.dart (hunk 1 above)
Comment (jason-simmons) [inline examples/stocks/lib/main.dart]: done

**Comment by jason-simmons** [comment]
Removed the args arguments from the Intl.message() calls

Which import path should be changed?

**Comment by alan-knight** [comment]
It looked like the Travis tests were failing because the import of ../i18n/stock_messages_all.dart wasn't found. That path looks right to me from a superficial look, so I'm not sure how the path resolution wants to work there.

**Comment by jason-simmons** [comment]
Our Travis environment is running dartanalyzer on the Flutter code.  The dartanalyzer is failing to resolve the "../i18n" path.  Moving i18n/ under lib/ works though.

After making that change, dartanalyzer still produces warnings about the auto-generated stock_messages_*.dart files.  Some of the warnings involve behavior that is safe at run time but that can not be statically verified.

**Comment by Hixie** [comment]
Those all seem fixable, we should be able to fix the intl library to generate clean code.

**Comment by eseidelGoogle** [comment]
I recommend landing first and worrying about intls code cleanliness at a
later juncture. :)

**Comment by Hixie** [comment]
We shouldn't land it while it makes travis red. But we can hack the flutter tool to exclude the autogenerated files (e.g. by ignoring any warnings in any files that start with "this file was autogenerated" or whatever intl's tool outputs).

It would be great if you could extend this example (possibly in a future PR) to some more complicated issues, like inserting numbers into strings where in some languages you need different strings for "zero", "one", "two", and higher numbers (just like in english you need different strings for singular and plural); or cases like ordinals ("1st", "2nd", etc); date and time formatting.

**Comment by Hixie** [comment]
I'll take care of the analyzer changes.

**Comment by Hixie** [comment]
Once https://github.com/flutter/flutter/pull/719 lands, rerun the travis check.

**Comment by Hixie** [comment]
Ok, the intl autogenerated files are ignored now. The remaining travis redness is real.

**Comment by alan-knight** [comment]
One of my items in the next while is to make the Intl package strong-mode clean. And another is to put in a transformer, which can make the generated code better if we can modify the original call-site. However, given that what this is fundamentally doing is looking up a function in a map keyed by string name, then applying it with whatever number of arguments (with varying types, mostly strings, but plurals can have numbers and it probably ought to allow enums for Intl.select) the user provided, fully statically verifiable behavior seems tricky.

For numbers and dates, you need to specify a format and format them yourself before passing to the message. So it would be something like the following.

```
youveGotMail(int messages) {
  // Except that the format is probably from an outer scope, they're inefficient to create 
  // and you don't want this in  every message that passes numbers. 
  // You can also give it an explicit pattern and/or locale.
  var f = new NumberFormat.DecimalPattern();
  return youveGotMailMessage(f.format(messages)); 
}

youveGotMailMessage(String numberOfMessages) => "You have $numberOfMessages new messages";

**Comment by jason-simmons** [comment]
Moved i18n/ under lib/.  With Ian's changes, this is now passing on Travis

**Pull request merged**